File: solutions/ocaml/leap/2/leap.ml
let leap_year year =
  let divisible_by4 = year mod 4 = 0 in
  let divisible_by100 = year mod 100 = 0 in
  let divisible_by400 = year mod 400 = 0 in

  divisible_by400 || (divisible_by4 && not divisible_by100)

File: solutions/python/leap/2/leap.py
"""Determine if a year is leap."""


def leap_year(year: int) -> bool:
    """Returns true if the passed year is leap."""
    divisible_by4 = year % 4 == 0
    divisible_by100 = year % 100 == 0
    divisible_by400 = year % 400 == 0

    return divisible_by400 or (divisible_by4 and not divisible_by100)
